src/main/kotlin/com/mdrobnak/lalrpop/inspections/MissingTerminalsInspection.kt

```diff
@@ -7,48 +7,61 @@ import com.intellij.codeInspection.ProblemsHolder
 import com.intellij.psi.PsiElement
 import com.intellij.psi.PsiElementVisitor
 import com.mdrobnak.lalrpop.psi.*
+import java.util.*
 
 object MissingTerminalsInspection : LocalInspectionTool() {
-    var unresolved = mutableListOf<PsiElement>()
-    var terminalDefs = mutableListOf<PsiElement>()
+    // null if there is an _ in the match block
+    var unresolved: MutableList<PsiElement>? = null
+    lateinit var terminalDefs: MutableList<PsiElement>
+
+    // true if it found an enum / match token
+    var hasTerminalDefsBlock = false
```

Comment:
Assignment (changing the value of a boolean or setting a variable to null) is not thread-safe unless it's wrapped in a synchronized block or similar primitive. The places where the boolean value is changed could be wrapped in a `synchronized` block to make it thread-safe.

We can generalize the boolean to act as an on/off switch for the inspection, so we don't have to set `unresolved` to null (instead set the boolean to `false`).